### ISSUE TITLE
Source map "sources" array should be resolved relative to source map

### ISSUE THREAD

**Issue by nex3** [postcss/postcss#1348]
Currently, if a previous source map's `"sources"` array contains relative URLs, those URLs are interpreted as relative to the current working directory rather than the location of the source map. This produces incorrect original file paths, which is particularly problematic in error messages. In conjunction with #1347, this makes it essentially impossible to correctly handle source map paths for CSS files outside the current working directory.

```js
// test.js
const postcss = require('postcss');
const fs = require('fs');

const root = postcss.parse(
  fs.readFileSync('subdir/test.css', 'utf8'), {from: 'subdir/test.css'});
console.log(root.source.input.origin(1, 1));
```

```shellsession
$ echo 'a {b: c}' > test.scss
$ mkdir subdir
$ sass test.scss subdir/test.css
$ pwd
/home/nweiz/goog/postcss
$ node test.js
{ file: '/home/nweiz/goog/test.scss', line: 1, column: 0 }
```

Note that the file path is `/home/nweiz/goog/test.scss`, while it should be `/home/nweiz/goog/postcss/test.scss`.

**Comment by evilebottnawi** [comment]
I think will be great to add tests for absolute `sources` too, `sass` can generate absoute `sources`, so merging them should work, maybe already tests, just want to say it

**Comment by ai** [comment]
@nex3 fixed 7ae3421

@evilebottnawi we already use absolute paths in `opts.map.absolute` test. Feel free to suggest some additional test.
